dolphinscheduler-storage-plugin/dolphinscheduler-storage-hdfs/src/main/java/org/apache/dolphinscheduler/plugin/storage/hdfs/HdfsStorageOperator.java

```diff
@@ -232,7 +232,11 @@ public List<StorageEntity> listFileStorageEntityRecursively(String resourceAbsol
 
         while (!foldersToFetch.isEmpty()) {
             String absolutePath = foldersToFetch.pollFirst();
-            RemoteIterator<LocatedFileStatus> remoteIterator = fs.listFiles(new Path(absolutePath), true);
+            Path path = new Path(absolutePath);
+            if (!fs.exists(path)) {
+                return result;
```

Comment:
`continue` is better.
```suggestion
                continue;
```

Reply:
Fixed.